lib/streamlit/elements/map.py

```diff
@@ -157,13 +158,20 @@ def _get_zoom_level(distance: float) -> int:
 
 
 def to_deckgl_json(data: Data, zoom: Optional[int]) -> str:
-    # TODO(harahu): The ignore statement here is because iterables don't have
-    #  the empty attribute. This is either a bug, or the documented data type
-    #  is too broad. One or the other should be addressed, and the ignore
-    #  statement removed.
-    if data is None or data.empty:  # type: ignore[union-attr]
+    if data is None:
         return json.dumps(_DEFAULT_MAP)
 
+    if hasattr(data, "empty"):
```

Comment:
I think this can be simplified a bit:

```python
if hasattr(data, "empty") && data.empty:
    return json.dumps(_DEFAULT_MAP)
```

And I think we can remove the ignore and the comment from harahu.

Reply:
Ok, so version with `and` requires `# type: ignore`, which makes @harahu comment completely up to date and valid I think 🤔 

Also you meant `and` right? Not `&` operator?

I've updated the PR with valid example

Reply:
Ahh yes, meant `and`. Ok, in that case we can leave the comment. I that `hasattr(data, "empty")` would give enough hints to the typing tool for this 🤔